### ISSUE TITLE
Test flake (?): MAC address 'A8:40:25:FF:A1:63' is not available

### ISSUE THREAD

**Issue by jgallagher** [oxidecomputer/omicron#3577]
On #3443 which makes no code changes, we saw this test failure: https://buildomat.eng.oxide.computer/wg/0/details/01H55DDK02KNS0KW2NCH10X1YC/ttvtOx1PkqfDMi3BpCDiV2kXoQGLFKR3GEEYq2chHz3odTm7/01H55DER4ZTMT973P4GDN6WT62

```
5570 | 2023-07-12T18:04:11.284Z | ---- integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first stdout ----
-- | -- | --
5571 | 2023-07-12T18:04:11.289Z | log file: /var/tmp/omicron_tmp/test_all-f28450499ad865b3-test_multiple_layers_of_snapshots_delete_all_snapshots_first.7612.206.log
5572 | 2023-07-12T18:04:11.298Z | note: configured to log to "/var/tmp/omicron_tmp/test_all-f28450499ad865b3-test_multiple_layers_of_snapshots_delete_all_snapshots_first.7612.206.log"
5573 | 2023-07-12T18:04:11.306Z | DB URL: postgresql://root@[::1]:46792/omicron?sslmode=disable
5574 | 2023-07-12T18:04:11.313Z | DB address: [::1]:46792
5575 | 2023-07-12T18:04:11.319Z | thread 'integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first' panicked at 'Could not initialize rack: InvalidRequest { message: "The MAC address 'A8:40:25:FF:A1:63' is not available" }', nexus/src/lib.rs:254:14
5576 | 2023-07-12T18:04:11.325Z | stack backtrace:
5577 | 2023-07-12T18:04:11.333Z | 0: rust_begin_unwind
5578 | 2023-07-12T18:04:11.340Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
5579 | 2023-07-12T18:04:11.345Z | 1: core::panicking::panic_fmt
5580 | 2023-07-12T18:04:11.350Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
5581 | 2023-07-12T18:04:11.353Z | 2: core::result::unwrap_failed
5582 | 2023-07-12T18:04:11.356Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/result.rs:1687:5
5583 | 2023-07-12T18:04:11.359Z | 3: core::result::Result<T,E>::expect
5584 | 2023-07-12T18:04:11.363Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/result.rs:1046:23
5585 | 2023-07-12T18:04:11.366Z | 4: <omicron_nexus::Server as nexus_test_interface::NexusServer>::start::{{closure}}
5586 | 2023-07-12T18:04:11.369Z | at ./src/lib.rs:235:9
5587 | 2023-07-12T18:04:11.372Z | 5: <core::pin::Pin<P> as core::future::future::Future>::poll
5588 | 2023-07-12T18:04:11.375Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/future/future.rs:125:9
5589 | 2023-07-12T18:04:11.378Z | 6: nexus_test_utils::ControlPlaneTestContextBuilder<N>::start_nexus_external::{{closure}}
5590 | 2023-07-12T18:04:11.381Z | at ./test-utils/src/lib.rs:558:9
5591 | 2023-07-12T18:04:11.384Z | 7: nexus_test_utils::test_setup_with_config::{{closure}}
5592 | 2023-07-12T18:04:11.387Z | at ./test-utils/src/lib.rs:756:9
5593 | 2023-07-12T18:04:11.390Z | 8: nexus_test_utils::test_setup::{{closure}}
5594 | 2023-07-12T18:04:11.409Z | at ./test-utils/src/lib.rs:141:5
5595 | 2023-07-12T18:04:11.413Z | 9: test_all::integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first::{{closure}}
5596 | 2023-07-12T18:04:11.418Z | at ./tests/integration_tests/volume_management.rs:866:1
5597 | 2023-07-12T18:04:11.423Z | 10: <core::pin::Pin<P> as core::future::future::Future>::poll
5598 | 2023-07-12T18:04:11.426Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/future/future.rs:125:9
5599 | 2023-07-12T18:04:11.430Z | 11: <core::pin::Pin<P> as core::future::future::Future>::poll
5600 | 2023-07-12T18:04:11.434Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/future/future.rs:125:9
5601 | 2023-07-12T18:04:11.437Z | 12: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}::{{closure}}
5602 | 2023-07-12T18:04:11.440Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:651:57
5603 | 2023-07-12T18:04:11.444Z | 13: tokio::runtime::coop::with_budget
5604 | 2023-07-12T18:04:11.447Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/coop.rs:107:5
5605 | 2023-07-12T18:04:11.452Z | 14: tokio::runtime::coop::budget
5606 | 2023-07-12T18:04:11.456Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/coop.rs:73:5
5607 | 2023-07-12T18:04:11.458Z | 15: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}
5608 | 2023-07-12T18:04:11.462Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:651:25
5609 | 2023-07-12T18:04:11.465Z | 16: tokio::runtime::scheduler::current_thread::Context::enter
5610 | 2023-07-12T18:04:11.468Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:410:19
5611 | 2023-07-12T18:04:11.470Z | 17: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}
5612 | 2023-07-12T18:04:11.473Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:650:36
5613 | 2023-07-12T18:04:11.476Z | 18: tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}
5614 | 2023-07-12T18:04:11.479Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:729:68
5615 | 2023-07-12T18:04:11.482Z | 19: tokio::runtime::context::scoped::Scoped<T>::set
5616 | 2023-07-12T18:04:11.485Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/context/scoped.rs:40:9
5617 | 2023-07-12T18:04:11.488Z | 20: tokio::runtime::context::set_scheduler::{{closure}}
5618 | 2023-07-12T18:04:11.491Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/context.rs:176:26
5619 | 2023-07-12T18:04:11.494Z | 21: std::thread::local::LocalKey<T>::try_with
5620 | 2023-07-12T18:04:11.497Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/thread/local.rs:252:16
5621 | 2023-07-12T18:04:11.500Z | 22: std::thread::local::LocalKey<T>::with
5622 | 2023-07-12T18:04:11.503Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/thread/local.rs:228:9
5623 | 2023-07-12T18:04:11.506Z | 23: tokio::runtime::context::set_scheduler
5624 | 2023-07-12T18:04:11.509Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/context.rs:176:9
5625 | 2023-07-12T18:04:11.512Z | 24: tokio::runtime::scheduler::current_thread::CoreGuard::enter
5626 | 2023-07-12T18:04:11.514Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:729:27
5627 | 2023-07-12T18:04:11.517Z | 25: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
5628 | 2023-07-12T18:04:11.521Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:638:19
5629 | 2023-07-12T18:04:11.524Z | 26: tokio::runtime::scheduler::current_thread::CurrentThread::block_on::{{closure}}
5630 | 2023-07-12T18:04:11.526Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:175:28
5631 | 2023-07-12T18:04:11.529Z | 27: tokio::runtime::context::runtime::enter_runtime
5632 | 2023-07-12T18:04:11.532Z | error: test failed, to rerun pass `-p omicron-nexus --test test_all`
5633 | 2023-07-12T18:04:11.535Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/context/runtime.rs:65:16
5634 | 2023-07-12T18:04:11.538Z | Running `/work/oxidecomputer/omicron/target/debug/deps/omicron_package-430baa7ce20a3752`
5635 | 2023-07-12T18:04:11.540Z | 28: tokio::runtime::scheduler::current_thread::CurrentThread::block_on
5636 | 2023-07-12T18:04:11.543Z | Running `/work/oxidecomputer/omicron/target/debug/deps/omicron_package-bff2457adf731b5c`
5637 | 2023-07-12T18:04:11.545Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/scheduler/current_thread.rs:167:9
5638 | 2023-07-12T18:04:11.548Z | Running `/work/oxidecomputer/omicron/target/debug/deps/omicron_passwords-37e130f5f4d67ec8`
5639 | 2023-07-12T18:04:11.551Z | 29: tokio::runtime::runtime::Runtime::block_on
5640 | 2023-07-12T18:04:11.554Z | at /home/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.29.1/src/runtime/runtime.rs:311:47
5641 | 2023-07-12T18:04:11.557Z | 30: test_all::integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first
5642 | 2023-07-12T18:04:11.560Z | at ./tests/integration_tests/volume_management.rs:866:1
5643 | 2023-07-12T18:04:11.562Z | 31: test_all::integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first::{{closure}}
5644 | 2023-07-12T18:04:11.565Z | at ./tests/integration_tests/volume_management.rs:866:1
5645 | 2023-07-12T18:04:11.568Z | 32: core::ops::function::FnOnce::call_once
5646 | 2023-07-12T18:04:11.570Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
5647 | 2023-07-12T18:04:11.573Z | 33: core::ops::function::FnOnce::call_once
5648 | 2023-07-12T18:04:11.576Z | at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
5649 | 2023-07-12T18:04:11.579Z | note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
5650 | 2023-07-12T18:04:11.583Z | WARN: dropped CockroachInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
5651 | 2023-07-12T18:04:11.586Z | WARN: temporary directory leaked: /var/tmp/omicron_tmp/.tmp8dCe1x
5652 | 2023-07-12T18:04:11.588Z | WARN: dropped ClickHouseInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
5653 | 2023-07-12T18:04:11.591Z | WARN: dropped DendriteInstance without cleaning it up first (there may still be a child process running and a temporary directory leaked)
5654 | 2023-07-12T18:04:11.594Z | WARN: dendrite temporary directory leaked: /var/tmp/omicron_tmp/.tmpSES1bl
5655 | 2023-07-12T18:04:11.600Z |  
5656 | 2023-07-12T18:04:11.602Z |  
5657 | 2023-07-12T18:04:11.605Z | failures:
5658 | 2023-07-12T18:04:11.608Z | integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first
```

which I assume to be a flake based on the PR making no related changes. I'll attempt a retry while poking at this a bit.

**Comment by davepacheco** [comment]
The error seems to come from here:
https://github.com/oxidecomputer/omicron/blob/fb1d4521564351d0a87f10df6520891ee3cabf12/nexus/db-queries/src/db/queries/network_interface.rs#L371-L376

The only MAC addr I see in the rack initialization params is in the ServiceNic, which is part of some ServiceKind variants in the ServicePutRequests:
https://github.com/oxidecomputer/omicron/blob/fb1d4521564351d0a87f10df6520891ee3cabf12/nexus/types/src/internal_api/params.rs#L162

In the test suite, we generate these with `random_system()`:
https://github.com/oxidecomputer/omicron/blob/fb1d4521564351d0a87f10df6520891ee3cabf12/nexus/test-utils/src/lib.rs#L465

which generates it from the range a0:40:25:ff:7f:ff to a0:40:25:ff:ff:ff:
https://github.com/oxidecomputer/omicron/blob/fb1d4521564351d0a87f10df6520891ee3cabf12/common/src/api/external/mod.rs#L1891-L1906.

That's only 32K addresses.  Maybe we just got unlucky?  In this case the test suite could consider just allocating them serially from a hardcoded range.

**Comment by smklein** [comment]
It's not a particularly informative data-point, but I also saw this test appear to cause a hang in https://buildomat.eng.oxide.computer/wg/0/details/01H54JVVZK4T8ECQ2HXCCN9NSW/ORZDVfCIg1ab7EL9LEvcUQx8Px4ng9E9s5xj7cndJ3PHbzV1/01H55DPZWM4GRC2V8CFZBH8Z6Z:

```
6020	2023-07-12T17:03:45.780Z	test integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_disks_first ... ok
6021	2023-07-12T17:03:45.862Z	test integration_tests::unauthorized::test_unauthorized has been running for over 60 seconds
6022	2023-07-12T17:03:49.398Z	test integration_tests::unauthorized::test_unauthorized ... ok
6023	2023-07-12T17:03:51.708Z	test integration_tests::volume_management::test_multiple_layers_of_snapshots_random_delete_order ... ok
6024	2023-07-12T17:03:55.350Z	test integration_tests::volume_management::test_multiple_snapshots ... ok
6025	2023-07-12T17:03:56.027Z	test integration_tests::volume_management::test_snapshot_prevents_other_disk ... ok
6026	2023-07-12T17:04:02.818Z	test integration_tests::volume_management::test_volume_checkout ... ok
6027	2023-07-12T17:04:05.070Z	test integration_tests::volume_management::test_volume_checkout_updates_multiple_gen ... ok
6028	2023-07-12T17:04:06.556Z	test integration_tests::volume_management::test_snapshot_then_delete_disk ... ok
6029	2023-07-12T17:04:08.533Z	test integration_tests::volume_management::test_volume_checkout_updates_nothing ... ok
6030	2023-07-12T17:04:09.120Z	test integration_tests::volume_management::test_volume_checkout_updates_sparse_mid_multiple_gen ... ok
6031	2023-07-12T17:04:15.974Z	test integration_tests::volume_management::test_volume_checkout_updates_sparse_multiple_gen ... ok
6032	2023-07-12T17:04:18.271Z	test integration_tests::volume_management::test_volume_hard_delete_idempotent ... ok
6033	2023-07-12T17:04:19.871Z	test integration_tests::volume_management::test_volume_remove_read_only_parent_bad_volume ... ok
6034	2023-07-12T17:04:21.922Z	test integration_tests::volume_management::test_volume_remove_read_only_parent_base ... ok
6035	2023-07-12T17:04:23.135Z	test integration_tests::volume_management::test_volume_remove_read_only_parent_no_parent ... ok
6036	2023-07-12T17:04:24.481Z	test integration_tests::volume_management::test_multiple_layers_of_snapshots_delete_all_snapshots_first has been running for over 60 seconds
6037	2023-07-12T17:04:30.134Z	test integration_tests::volume_management::test_volume_remove_read_only_parent_volume_deleted ... ok
6038	2023-07-12T17:04:32.272Z	test integration_tests::volume_management::test_volume_remove_read_only_parent_volume_not_volume ... ok
6039	2023-07-12T17:04:34.205Z	test integration_tests::volume_management::test_volume_remove_rop_saga ... ok
6040	2023-07-12T17:04:36.238Z	test integration_tests::volume_management::test_volume_remove_rop_saga_deleted_volume ... ok
6041	2023-07-12T17:04:37.585Z	test integration_tests::volume_management::test_volume_remove_rop_saga_no_volume ... ok
6042	2023-07-12T17:04:45.098Z	test integration_tests::volume_management::test_volume_remove_rop_saga_twice ... ok
6043	2023-07-12T17:04:46.411Z	test integration_tests::volume_management::test_volume_remove_rop_saga_volume_not_volume ... ok
6044	2023-07-12T17:04:51.940Z	test integration_tests::vpc_firewall::test_vpc_firewall ... ok
6045	2023-07-12T17:04:54.816Z	test integration_tests::vpc_routers::test_vpc_routers ... ok
6046	2023-07-12T17:04:58.084Z	test integration_tests::vpc_subnets::test_delete_vpc_subnet_with_interfaces_fails ... ok
6047	2023-07-12T17:05:03.397Z	test integration_tests::vpc_subnets::test_vpc_subnets ... ok
6048	2023-07-12T17:05:04.776Z	test integration_tests::zpools::test_zpool_put_bad_physical_disk_returns_not_found ... ok
6049	2023-07-12T17:05:06.841Z	test integration_tests::vpcs::test_vpcs ... ok
6050	2023-07-12T17:05:07.220Z	test integration_tests::zpools::test_zpool_put_bad_sled_returns_not_found ... ok
6051	2023-07-12T17:05:10.395Z	test integration_tests::zpools::test_zpool_put_success ... ok
 <note the jump in time>
6052	2023-07-12T18:39:37.299Z	job cancelled
```

**Comment by bnaecker** [comment]
I also saw this last night, similar behavior to Sean's. Recent changes seem to have made this more likely, but that could just be sampling bias.

**Comment by davepacheco** [comment]
I'm going to assume this is fixed by #3581.  It's possible it's not, that there's something else going on there, but we'll never be able to know that from the data we have so let's just reopen if we see this again.
